### DISCUSSION TITLE
Server down time

### DISCUSSION THREAD

**Discussion by Arash-72** [r-simmer/simmer#303] [Q&A] [answered]
I want to simulate a clinic in which a patient upon arrival visits a nurse for two minutes. Then visits the doctor for 3 minutes. Then again visits a nurse for 4 minutes and then leaves the clinic. However, the doctor right after visiting a patient doesn't take the next patient immediately and becomes unavailable to do some additional tasks (doing some documentation for that patient) for 1 minute. Such additional tasks don't have any effect on the leaving time of the current patient. 
Assume that I have a total of two patients with arrival times of 0 and 3. 
How can I include the doctor's unavailability time in the simulation in the correct way?

Here is the code that I have used clone(). 

```r
env <- simmer()

patient_trajectory <- trajectory() %>%
  set_attribute("arr", function() {now(env)}) %>% 
  seize("nurse") %>%
  timeout(2) %>% 
  release("nurse") %>% 
  seize("Doctor") %>%
  set_attribute("doctor busy start", function() {now(env)}) %>% 
  timeout(3) %>%
  set_attribute("doctor busy end", function() {now(env)}) %>% 
  release("Doctor") %>% 
  clone(2, trajectory() %>% seize("Doctor") %>%
          set_attribute("doctor unavailable start", function() {now(env)}) %>% 
          timeout(1) %>%
          set_attribute("doctor unavailable end", function() {now(env)}) %>% 
          release("Doctor")) %>% 
  seize("nurse") %>%
  timeout(4) %>% 
  release("nurse") %>% 
  set_attribute("patient exit", function() {now(env)})

  
  
env %>% 
  add_resource("Doctor", capacity = 1) %>% 
  add_resource("nurse", capacity = Inf) 

env %>%
  add_generator("Patient", patient_trajectory, at(0, 3),mon = 2) 


env %>% run(until = 200)
env %>% get_mon_attributes()
```

But, as shown in the attached photo, the issue is the doctor starts visiting patient 2 before becoming unavailable after visiting patient 1. 
![Sim](https://github.com/r-simmer/simmer/assets/107949364/eff18611-185f-480b-87e9-7838ea9b768c)

Thank you so much for your time and attention in advance.

**Comment by Enchufa2** [comment, marked as answer]
This pattern is a "delayed release": the patient leaves, but the resource is still busy for some time. We implemented a helper for this in `simmer.bricks`, see [`delayed_release`](https://r-simmer.org/extensions/bricks/reference/delayed_release). (You can see that this helper uses `clone` internally, so you were headed in the right direction, but some pieces were missing in your solution).

**Comment by Arash-72** [comment]
delayed_release exactly does what I had in mind.
Thank you so much.

**Comment by Arash-72** [comment]
Hi.

I was wondering if you also know how to capture the time that a resource becomes available after a delayed release with attributes.
Unfortunately, I can't get it by using set_attribute("doctor unavailable end", function() {now(env)}) inside the trajectory like below:

...
delayed_release("Doctor") %>% 
 set_attribute("doctor unavailable end", function() {now(env)}) %>% 
...

**Reply by Enchufa2** [reply]
This is how a `delayed_release` works: https://github.com/r-simmer/simmer.bricks/blob/bd348f1487deabe4c79a2efafaa22b1c041829cb/R/delayed_release.R#L51-L76. So you can just copy the function and rewrite it to add the attribute change in the proper place.

**Reply by Arash-72** [reply]
Thank you very much.
